dags/sentinel-2_gap_filler.py

```diff
@@ -0,0 +1,200 @@
+
+This DAG is intended to be run manually, and must be passed a configuration JSON object
+specifying `offset`, `limit` line numbers of the report and `s3_report_path` specifying
+the path to the gap report file.
+
+Eg:
+```json
+{"offset": 824, "limit": 1224, "s3_report_path": "s3://deafrica-sentinel-2/monthly-status-report/2020-11-11T01:38:02.023140.txt"}
+"""
+import json
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from datetime import datetime
+from typing import Dict
+
+from airflow import DAG
+from airflow.contrib.sensors.aws_sqs_sensor import SQSHook
+from airflow.hooks.S3_hook import S3Hook
+from airflow.operators.python_operator import PythonOperator
+
+SRC_BUCKET_NAME = "sentinel-cogs"
+QUEUE_NAME = "deafrica-prod-eks-sentinel-2-data-transfer"
+PRODUCT_NAME = "s2_l2a"
+
+default_args = {
+    "owner": "Airflow",
+    "start_date": datetime(2020, 7, 24),
+    "email": ["toktam.ebadi@ga.gov.au"],
+    "email_on_failure": True,
+    "email_on_retry": False,
+    "retries": 0,
+    "schedule_interval": "@once",
+    "us_conn_id": "prod-eks-s2-data-transfer",
+    "africa_conn_id": "deafrica-prod-migration",
+}
+
+
+def get_common_message_attributes(stac_doc: Dict) -> Dict:
+    """
+    Returns common message attributes dict
+    :param stac_doc: STAC dict
+    :return: common message attributes dict
+    """
+    msg_attributes = {}
+    product = PRODUCT_NAME
+    msg_attributes["product"] = {
+        "DataType": "String",
+        "StringValue": product,
+    }
+
+    datetime = stac_doc.get("properties").get("datetime")
+    if datetime:
+        msg_attributes["datetime"] = {
+            "DataType": "String",
+            "StringValue": datetime,
+        }
+
+    cloudcover = stac_doc.get("properties").get("eo:cloud_cover")
+    if cloudcover:
+        msg_attributes["cloudcover"] = {
+            "DataType": "Number",
+            "StringValue": str(cloudcover),
+        }
+
+    maturity = stac_doc.get("properties").get("dea:dataset_maturity")
+    if maturity:
+        msg_attributes["maturity"] = {
+            "DataType": "String",
+            "StringValue": maturity,
+        }
+
+    bbox = stac_doc.get("bbox")
+    if bbox and len(bbox) > 3:
+        msg_attributes["bbox.ll_lon"] = {
+            "DataType": "Number",
+            "StringValue": str(bbox[0]),
+        }
+        msg_attributes["bbox.ll_lat"] = {
+            "DataType": "Number",
+            "StringValue": str(bbox[1]),
+        }
+        msg_attributes["bbox.ur_lon"] = {
+            "DataType": "Number",
+            "StringValue": str(bbox[2]),
+        }
+        msg_attributes["bbox.ur_lat"] = {
+            "DataType": "Number",
+            "StringValue": str(bbox[3]),
+        }
+
+    return msg_attributes
+
+
+def get_missing_stac_files(s3_report_path, offset=0, limit=None):
+    """
+    read the gap report
+    """
+
+    hook = S3Hook(aws_conn_id=dag.default_args["africa_conn_id"])
```

Comment:
We can just use the Oregon hook to do this, I think.

Reply:
No, because the oregon hook has a 'us-west2' as the region